src/code/z_room.c

```diff
@@ -295,7 +295,7 @@ void Room_DrawBackground2D(Gfx** gfxP, void* tex, void* tlut, u16 width, u16 hei
     Room_DecodeJpeg(SEGMENTED_TO_VIRTUAL(tex));
 
     bg = (uObjBg*)(gfx + 1);
-    gSPBranchList(gfx, (u32)bg + sizeof(uObjBg));
+    gSPBranchList(gfx, (uintptr_t)bg + sizeof(uObjBg));
```

Comment:
```suggestion
    gSPBranchList(gfx, bg + 1);
```
These casts on `bg` are actually not needed at all thanks to pointer arithmetic, although it may make it slightly less clear at a glance.

Reply:
Imo the code is much clearer with the sizeof, and this code is already pretty low level so the casts seem fine to me, but it's probably just personal preference

Reply:
I think `gSPBranchList(gfx, (Gfx*)(bg + 1));` would be better but I don't care too much

Reply:
I also prefer just the pointer arithmetic. Also just for the sake of a slightly different perspective `&bg[1]`?

Reply:
Actually on second thought Dragorn's suggestion seems good to me. Clear that it is going past the `bg` to grab the next thing as `Gfx`.

Reply:
I changed it to `bg + 1` since it seems to be the popular option (fun fact it's basically how I wrote it when I decompiled it, until it was changed to using sizeof at some point 😅)